### ISSUE TITLE
Become the "official" macOS port of the HomeworldSDL community?

### ISSUE THREAD

**Issue by ThibaultLemaire** [atsb/Hiigara#2]
Hi!

I'm the maintainer of the Tennhauser Gate fork, speaking for the reborn HomeworldSDL community.

A kind person made us aware of your fork, suggesting to list it as the macOS (and linux) fork: https://github.com/HomeworldSDL/HomeworldSDL/pull/12

1. I wanted to make sure it was OK by you before including a link to your repo
2. Before endorsing your fork I wanted to know if you would be willing to coordinate with us on developments

Having individual forks gives us the freedom to take things our way, but having some basic level of communication between forks let's everyone leverage the work of others.

The so-called reborn HWSDL community is currently composed of the Gardens of Kadesh maintainer, and myself, with the blessing of some previous maintainers like Ted Cippichio, support and feedback from some homeworld modders, and occasional help from people coming and going.

Our long-term plan is to merge all the bugfixing we have been doing into the main HomeworldSDL/HomeworldSDL repo, to provide the most stable UX across all platforms.

So... "What are your intentions?" :)

**Comment by atsb** [comment]
That is fine by me and I am glad the community is doing this.  Less ports and more cohesion between maintainers is a great thing for both developers and users.  Since Hiigara is a modified version of Garden of Kadesh codebase, working with you in this will be a sensibel choice and also a somewhat simplified one due to the similarity between the forks.

Intention wise, I aim for a stable, easy to use version for Linux and macOS.

**Comment by ThibaultLemaire** [comment]
Awesome then!

Most of our discussion happens on discord, but we also have a matrix space if you would like to join us!

Regarding the "official" ports, there hadn't been the case so far that two forks overlapped in their focused platforms, so... I'm not sure how to handle that. I'd say having only one designated fork for a given platform would be best. Or maybe highlight one as "recommended".

I also aim for stability on TG and I'm currently working on a bugfix release cherry-picking stuff from GoK (I'm tracking progress here: https://gitlab.com/homeworldsdl/TenhauserGate/-/issues/36). I am voluntarily not including much of the work that happened on GoK (yet) as that makes the game run noticeably slower on my old CPU (stuff like the universe update rate increase, rewrite of the OpenGL calls, and merging of the audio thread into the main thread).

I don't know where you are with Hiigara (you don't seem to have kept the git history), so now that you know a bit more of the situation on TG, what is your opinion on which one of the two should be recommended for Linux users? (I will also try to compile your fork to see how it goes)

P.S.: If you have bug fixes I could also pick from your fork feel free to recommend them!

**Comment by atsb** [comment]
Indeed, the history was lost when I created the github repo unfortunately.  I forked it from the last stable 'development' version of Garden of Kadesh (roughly a year ago, around September 2022).  The commits here represent the fixes I did to prevent random crashing on macOS as well as on Linux, there are a few more little bits but nothing hugely critical, mostly macros for __APPLE__ and project fixes.  After that, I kept it intentionally stable.

About the overlaps, I also think it is best to have only one port per platform.  No issues with me at all, only that in regards to OpenGL, I would be obviously limited to around OpenGL 4.x due to Apple not providing modern versions anymore, so any updates ideally would center around OpenGL 3.3 or 4.0 for maximum compatibility.

I'd be happy to join the discord, link?
